src/server/controllers/user-profile.controller.ts

```diff
@@ -52,3 +57,49 @@ export const updateUserProfileHandler = async ({
     throw throwDbError(error);
   }
 };
+export const addEntityToShowcaseHandler = async ({
+  input,
+  ctx,
+}: {
+  input: ShowcaseItemSchema;
+  ctx: DeepNonNullable<Context>;
+}) => {
+  try {
+    if (input.entityType !== 'Model' && input.entityType !== 'Image') {
+      throw new Error('Invalid entity type. Only models and images are supported right now');
+    }
+
+    if (input.entityType === 'Model') {
+      await dbRead.model.findUniqueOrThrow({
+        where: { id: input.entityId },
+      });
+    }
+
+    if (input.entityType === 'Image') {
+      await dbRead.image.findUniqueOrThrow({
+        where: { id: input.entityId },
+      });
+    }
```

Comment:
I would avoid making db calls in the controller, I know we do it in other places, but probably makes sense to do these checks inside the service anyways